### DISCUSSION TITLE
Maximum distance accepted by the renderer

### DISCUSSION THREAD

**Discussion by Ketaries** [mitsuba-renderer/mitsuba3#367] [Q&A] [answered]
Hi all,

I'm working with mitsuba for research purposes and I need to render scenes with real world distances which are quite big (~10 000 km). In order to select a proper ratio between real world units and mitsuba world units I need to know if mitsuba has a maximum limit for defining the position of objects or their size.

Thank you in advance !

**Comment by Ketaries** [comment, marked as answer]
Looks like the solution was simply to modify the `far_clip` value when defining the sensor in order to extend the limit of the scene. I missed that in the doc before, my bad !

**Reply by njroussel** [reply]
At your scale, one other important thing that you might want do to, if you haven't already, is to use a `double` [variant](https://mitsuba.readthedocs.io/en/stable/src/key_topics/variants.html). We don't ship any `double` variants in the Python packages, so you would need to build the project yourself.
You would also need to use the built-in KD-Tree, by turning off the `MI_ENABLE_EMBREE` option (it's a CMake option). This comes as a significant performance cost, but it is the only solution we currently provide if you need double-precision raytracing acceleration structures.

**Reply by Ketaries** [reply]
Oh indeed, I'll check if it works as it is for now otherwise I'll do that. Thank you for the tips !
